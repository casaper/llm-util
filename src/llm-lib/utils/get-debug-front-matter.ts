import { ChatResponse } from 'ollama';
import { stringify } from 'yaml';

export const getDebugFrontMatter = (
  debug: boolean,
  {
    model,
    eval_count,
    prompt_eval_count,
    total_duration,
    eval_duration,
    load_duration,
    prompt_eval_duration,
    created_at,
  }: ChatResponse
) =>
  debug
    ? `---\n${stringify({
        model,
        eval_count,
        prompt_eval_count,
        eval_duration,
        load_duration,
        prompt_eval_duration,
        total_duration,
        created_at,
      })}---\n\n`
    : '';
