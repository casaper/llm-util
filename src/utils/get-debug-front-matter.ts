import { ChatResponse } from 'ollama';

import { createFrontMatter } from './create-front-matter';

/**
 * If debug is true a Markdown YAML front matter block is returned with some ChatResponse metadata.
 */
export const getDebugFrontMatter = (debug: boolean, response: ChatResponse) =>
  debug ? createFrontMatter(response) : '';
