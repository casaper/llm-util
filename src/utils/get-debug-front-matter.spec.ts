import { ChatResponse } from 'ollama';
import { describe, expect, it } from 'vitest';

import { getDebugFrontMatter } from './get-debug-front-matter';

describe('getDebugFrontMatter', () => {
  const mockChatResponse = {
    model: 'llama3.1:8b',
    created_at: new Date('2024-01-01T00:00:00Z'),
    eval_count: 100,
    prompt_eval_count: 200,
    total_duration: 123400,
    eval_duration: 56700,
    load_duration: 89000,
    prompt_eval_duration: 12300,
    message: {
      role: 'assistant',
      content: 'This is a test response.',
    },
    done: true,
    done_reason: 'done',
  } satisfies ChatResponse;

  it('should return an empty string when debug is false', () => {
    expect(getDebugFrontMatter(mockChatResponse, false)).toBe('');
  });

  it('should return an empty string when debug is undefined', () => {
    expect(getDebugFrontMatter(mockChatResponse, undefined)).toBe('');
  });

  it('should return debug front matter when debug is true', () => {
    const result = getDebugFrontMatter(mockChatResponse, true);
    expect(result).toMatchSnapshot();
  });
});
