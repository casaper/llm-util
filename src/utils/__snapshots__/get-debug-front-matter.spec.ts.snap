// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`getDebugFrontMatter > should return debug front matter when debug is true 1`] = `
"---
model: llama3.1:8b
eval_count: 100
prompt_eval_count: 200
total_duration: 2m 3s 400ms
eval_duration: 56s 700ms
load_duration: 1m 29s
prompt_eval_duration: 12s 300ms
created_at: 2024-01-01T00:00:00.000Z
---

"
`;
